data_tools/firebase_client_wrapper.py

```diff
@@ -303,10 +303,18 @@ def delete_dataset(dataset_id):
 def _delete_collection(coll_ref, batch_size):
     docs = coll_ref.limit(batch_size).get()
     deleted = 0
+    batch = client.batch()
+    first_id = None
+    last_id = None
     for doc in docs:
-        print(u'Deleting doc {} => {}'.format(doc.id, doc.to_dict()))
-        doc.reference.delete()
+        if first_id is None:
+            first_id = doc.id
+        last_id = doc.id
+
+        batch.delete(doc.reference)
         deleted = deleted + 1
+    batch.commit()
```

Comment:
I think that this creates a requirement that batch_size < firebase's commit batch limit. We could probably extract this into a global constant to stop someone having to worry about it.

Reply:
What exactly is the suggestion here - removing the batch_size parameter from this function's signature and always using the constant, or keeping the parameter but using the constant at each call site?